modules/eval_php.py

```diff
@@ -0,0 +1,34 @@
+import html, json, socket, urllib
+from src import ModuleManager, utils
+
+
+EVAL_URL = "https://tools.tutorialspoint.com/execute_new.php"
+class Module(ModuleManager.BaseModule):
+    _name = "PHP"
+
+    def _evalphp(self, lang, event):
+        page = None
+        try:
+            page = utils.http.get_url(EVAL_URL,
+                post_data={
+                    "lang": lang,
+                    "code": event["args"],
+                    "execute": "%s main.php" % lang,
+                    "mainfile": "main.php",
+                    "stdinput": event["args"]
+                },
+                method="POST")
+        except:
+            pass
+
+        if page:
+            out = {}
+            out["out"] = page
+
+            event["stdout"].write(out["out"].strip("\n"))
+        else:
+            event["stderr"].write("%s: failed to eval" % event["user"].nickname)
+
+    @utils.hook("received.command.php", min_args=1)
+    def chooselang(self, event):
```

Comment:
This abstraction seems unnecessary, given this module only takes PHP

Reply:
Seconded on this - don't need two functions here. Should just `_evalphp` -> `eval` and have that have the `@utils.hook` on it rather than 2 functions